nl/nl_linux_test.go

```diff
@@ -75,7 +75,7 @@ func TestIfSocketCloses(t *testing.T) {
 		for {
 			_, _, err := sk.Receive()
 			// Receive returned because of a timeout and the FD == -1 means that the socket got closed
-			if err == unix.EAGAIN && nlSock.GetFd() == -1 {
+			if nlSock.GetFd() == -1 {
```

Comment:
The err is no longer `unix.EAGAIN` because `Receive` doesn't return when blocking on `unix.Recvfrom`, instead, it blocks on `unix.Poll`, then after 2s timeout, it calls `unix.Recvfrom` and gets "bad file descriptor"